db/migrate/20150313212337_refactor_saved_regressions.rb

```diff
@@ -0,0 +1,114 @@
+class RefactorSavedRegressions < ActiveRecord::Migration
+  def change
+    say 'Refactoring saved regressions'
+    Visualization.find_each do | v |
+      globals = JSON.parse(v.globals)
+      scatter_params = globals['Scatter']
+      timeline_params = globals['Timeline']
+      if !scatter_params.nil? and scatter_params.key? 'savedRegressions'
+        scatter_regressions = []
+        scatter_params['savedRegressions'].each do |regression|
+          keys = regression.keys
+          if keys.include? 'fieldIndices' and keys.include? 'series'
+            x_axis = regression['fieldIndices'][0]
+            y_axis = regression['fieldIndices'][1]
+            groups = regression['fieldIndices'][2]
+            id = regression['series']['name']['id']
+            dash_style = regression['series']['dashStyle']
+            name = regression['series']['name']['group']
+            r2 = regression['series']['name']['regression']['tooltip'].split('</strong> ')[3].gsub('e', 'E')
+            type = regression['type']
+            wtf = regression['series']['name']['regression']['tooltip'].split('<br>')[1].delete('^0-9 \.\-eE').split('  ').reverse
```

Comment:
lololol @ variable but 'whata terrible failure' should probably not be in our code